intelmq/bots/experts/cymru_whois/_lib.py

```diff
@@ -168,6 +168,8 @@ def __asn_query_parse(text):
         if items[4]:
             # unicode characters need to be decoded explicitly
             # with the help of https://stackoverflow.com/questions/60890590/
-            result['as_name'] = items[4].encode('latin1').decode('utf8')
+            as_name = items[4].encode('latin1', errors='ignore').decode('utf8')
+            if as_name:
```

Comment:
is this the "Avoid extraneous search domain-based queries on NXDOMAIN result"?

Reply:
That's the added periods in `IP_QUERY` and `ASN_QUERY`, actually.
Example: if you have a search domain of "lan" and resolving `homeserver` fails with NXDOMAIN, then the search domain will cause `homeserver.lan.` to be queried.
However, if, for instance, `1.2.3.4.origin4.cymru.com` returns NXDOMAIN, then `1.2.3.4.origin4.cymru.com.lan.` would be queried.
But, the added period at the end signifies that this is a fully qualified domain name and such queries should not be made.